src/Core/Form/IdentifiableObject/CommandBuilder/Product/ProductCategoriesCommandsBuilder.php

```diff
@@ -56,9 +56,8 @@ public function buildCommands(ProductId $productId, array $formData): array
         $associatedCategoryIds = [];
         $defaultCategoryId = 0;
         foreach ($productCategories as $categoryId => $categoryData) {
-            if ((bool) $categoryData['is_associated']) {
-                $associatedCategoryIds[] = (int) $categoryId;
-            }
+            $associatedCategoryIds[] = (int) $categoryId;
+
             if ((bool) $categoryData['is_default']) {
```

Comment:
Once the FormType has been modified with the new selector this will have to be updated and use `$formData['description']['categories']['default_category']` instead